### PR TITLE
[Mail] source field: make history clickable

### PR DESCRIPTION
**Pull request by atp-odoo**
Task: https://www.odoo.com/web#id=19637&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
pad: https://pad.odoo.com/p/r.xrbkexmZvkwd7Sp7

Make source document in the chatter and make clickable  too.

### CODE REVIEW

File: addons/mail/models/mail_thread.py [hunk 1]
```diff
@@ -1713,6 +1713,9 @@ def message_post(self, body='', subject=None, message_type='notification',
         # Post the message
         new_message = MailMessage.create(values)
 
+        if references:
+            new_message.write({'reference_ids': map(lambda x: (0, 0, x), references)})
```
Comment (atp-odoo) [inline addons/mail/models/mail_thread.py]: cannot pass referecens in values dictionary before message creation ?

File: addons/mail/security/ir.model.access.csv [hunk 2]
```diff
@@ -30,3 +30,7 @@ access_mail_template,mail.template,model_mail_template,base.group_user,1,1,1,0
 access_mail_template_system,mail.template_system,model_mail_template,base.group_system,1,1,1,1
 access_mail_shortcode,mail.shortcode,model_mail_shortcode,base.group_user,1,1,1,1
 access_mail_shortcode_portal,mail.shortcode.portal,model_mail_shortcode,base.group_portal,1,0,0,0
+access_mail_message_reference_all,mail.message.reference.all,model_mail_message_reference,,1,0,0,0
+access_mail_message_reference_portal,mail.message.reference.portal,model_mail_message_reference,base.group_portal,1,1,1,1
```
Comment (atp-odoo) [inline addons/mail/security/ir.model.access.csv]: portal users have not all rights.. only read access required.

File: addons/mail/models/mail_message_reference.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+
+from openerp import api, fields, models
```
Comment (atp-odoo) [inline addons/mail/models/mail_message_reference.py]: now use from odoo import api, fields, models ;)

File: addons/mail/models/mail_message_reference.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class MessageReference(models.Model):
+    _name = 'mail.message.reference'
```
Comment (atp-odoo) [inline addons/mail/models/mail_message_reference.py]: better to use _description also...

File: addons/mail/models/mail_thread.py (hunk 1 above)
Comment (rpi-odoo) [inline addons/mail/models/mail_thread.py]: Done :+1:

**Comment by hmo-odoo** [comment]
cancelled task

**Pull request closed without merging**